### PR TITLE
Resolve open Alphabet TODOs

### LINKED ISSUE #1268: Alphabet module TODOs
- [ ] enforce that underlying_char_t is a builtin_character and document this
  - [ ] decide whether to move over alphabet related views
  - [ ] debug and/or report original proxy implementation that causes ICEs

### PR DESCRIPTION
**Pull request by h-2**
fixes #1268 

I recommend reviewing commit-wise.

### CODE REVIEW

File: include/seqan3/alphabet/alphabet_base.hpp [hunk 1]
```diff
@@ -180,43 +180,37 @@ class alphabet_base
     //!\brief Checks whether the letters `lhs` and `rhs` are equal.
     friend constexpr bool operator==(derived_type const lhs, derived_type const rhs) noexcept
     {
-        using seqan3::to_rank;
-        return to_rank(lhs) == to_rank(rhs);
+        return seqan3::to_rank(lhs) == seqan3::to_rank(rhs);
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/alphabet/alphabet_base.hpp]: Within the scope of.the class there is also a member function of the same name (which is preferred if unqualified lookup happens), but we explicitly want the free function / CPO!

[the remark this answers is not in the document]
Comment (joergi-w) [reply include/seqan3/alphabet/alphabet_base.hpp]: Okay, thanks!

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp [hunk 2]
```diff
@@ -128,15 +53,23 @@ namespace seqan3
...
+                                            detail::valid_template_spec_or_t<void, alphabet_char_t, alphabet_type>>>
 {
 private:
     //!\brief Type of the base class.
-    using base_t =  alphabet_base<derived_type,
-                                  alphabet_size<alphabet_type>,
-                                  detail::valid_template_spec_or_t<void, alphabet_char_t, alphabet_type>>;
+    using base_t =
+        std::conditional_t<std::is_class_v<alphabet_type>,
+                           alphabet_type,                                   // inherit from emulated type if possible
```
Comment (smehringer) [inline include/seqan3/alphabet/detail/alphabet_proxy.hpp]: maybe add those comments above where the conditional appears first :nail_care:

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp [hunk 3]
```diff
@@ -247,17 +184,40 @@ class alphabet_proxy : public alphabet_base<derived_type,
     //!\brief Implicit conversion to the emulated type.
     constexpr operator alphabet_type() const noexcept
     {
-        return assign_rank_to(to_rank(), alphabet_type{});
+        if constexpr (std::is_class_v<alphabet_type>)
+            return *this;
+        else
+            return assign_rank_to(base_t::to_rank(), alphabet_type{});
```
Comment (smehringer) [inline include/seqan3/alphabet/detail/alphabet_proxy.hpp]: seqan3::assign_rank_to :point_up:

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp [hunk 4]
```diff
@@ -247,17 +184,40 @@ class alphabet_proxy : public alphabet_base<derived_type,
...
+
+        /* This is used explicitly in this class often to prevent considering alphabet_type's constructors
```
Comment (smehringer) [inline include/seqan3/alphabet/detail/alphabet_proxy.hpp]: This sentence makes no sense to me oO
Also, why don't you put this into the details of the documentation of the implicit conversion?

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp [hunk 5]
```diff
@@ -247,17 +184,40 @@ class alphabet_proxy : public alphabet_base<derived_type,
...
+         * during a static_cast.
+         *
+         * This is one of error cases:
+         * The tuple composite qualified returns a component_proxy which inherits from alphabet_proxy_base.
```
Comment (smehringer) [inline include/seqan3/alphabet/detail/alphabet_proxy.hpp]: `seqan3::`qualified

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp [hunk 6]
```diff
@@ -247,17 +184,40 @@ class alphabet_proxy : public alphabet_base<derived_type,
...
     //!\brief Implicit conversion to types that the emulated type is convertible to.
     template <typename other_t>
     //!\cond
-        requires std::convertible_to<alphabet_type, other_t>
+        requires (!std::is_class_v<alphabet_type>) && std::convertible_to<alphabet_type, other_t>
     //!\endcond
     constexpr operator other_t() const noexcept
     {
         return operator alphabet_type();
+
```
Comment (smehringer) [inline include/seqan3/alphabet/detail/alphabet_proxy.hpp]: remove extra newline

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp [hunk 7]
```diff
@@ -247,17 +184,40 @@ class alphabet_proxy : public alphabet_base<derived_type,
...
+    }
+
+    //!\brief Returns the character.
```
Comment (smehringer) [inline include/seqan3/alphabet/detail/alphabet_proxy.hpp]: the `rank` ~~character~~

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp (hunk 6 above)
Comment (smehringer) [reply include/seqan3/alphabet/detail/alphabet_proxy.hpp]: not sresolved

**Pull request merged**